### ISSUE TITLE
Auto format will remove all codes in current file

### ISSUE THREAD

**Issue by molinla** [biomejs/biome-intellij#107]
Overview
When checked `Run format on save` or `Run save fixed save` options, it will remove my current file whole code. i test on `.tsx`, `.ts`, `.css`. Their code will also be removed.

# Env Info
| Tag  | Version |
|  ----  | ----  |
| System  | Windows 11 23H2 |
| IDE  | Webstorm 2024.3 |
| BiomePlugin |  1.5.2 |
| pnpm | 9.12.1 |

# Bug Reproduction GIF Preview
> this gif wants to show when I type useless `whitespace` hope biome can remove useless `whitespace`, but it not remove useless `whitespace` it directly removes whole codes.

![PixPin_2024-12-04_14-12-24](https://github.com/user-attachments/assets/01f2873f-e0cf-44d7-884b-23eddee4ace5)

# Biome Config
```json5
{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "vcs": {
    "enabled": false,
    "clientKind": "git",
    "useIgnoreFile": false
  },
  "files": {
    "ignore": ["node_modules", "dist"]
  },
  "formatter": {
    "enabled": true,
    "indentStyle": "space"
  },
  "organizeImports": {
    "enabled": true
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "a11y": {
        "noSvgWithoutTitle": "warn",
        "useKeyWithClickEvents": "off",
        "useKeyWithMouseEvents": "off"
      },
      "correctness": {
        "useExhaustiveDependencies": "warn"
      },
      "style": {
        "useNodejsImportProtocol": "off",
        "noNonNullAssertion": "warn",
        "noParameterAssign": "warn"
      },
      "complexity": {
        "noForEach": "off"
      },
      "suspicious": {
        "noExplicitAny": "warn",
        "noAssignInExpressions": "off",
        "noConsole": {
          "level": "warn",
          "fix": "none"
        }
      },
      "performance": {
        "noAccumulatingSpread": "warn"
      }
    }
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single"
    },
    "parser": {
      "unsafeParameterDecoratorsEnabled": true
    }
  },
  /*
    TODO: remove it when biome support monorepo or plugin
    1. biome don't support monorepo: https://github.com/biomejs/biome/issues/2228
    2. must be disable useImportType in backend, cause biome don't support emitDecoratorMetadata,
        biome team also does not intend to support it, ts eslint also rejected in last version
        https://github.com/biomejs/biome/issues/2003#issuecomment-2028665621
    3. biome roadmap: https://github.com/biomejs/biome/issues/2455
  */
  "overrides": [
    {
      "include": [
        "apps/comment-service.app/**",
        "apps/crdt.app/**",
        "apps/copilot.app/**",
        "apps/user-login.app/**",
        "apps/oss-service.app/**",
        "apps/platform.app/**",
        "packages/platform/db/**",
        "packages/platform/ai-chat/**"
      ],
      "linter": {
        "rules": {
          "style": {
            "useImportType": "off"
          }
        }
      }
    }
  ]
}

```

# Related Info
I also tested on macbook air m1, with same webstorm version and same project. It works well.

**Comment by molinla** [comment]
# Resolved
i try to set biome.exe manually, its works. i guess it is cause
![image](https://github.com/user-attachments/assets/ae717b6a-0c22-4fae-a81f-090ee57722dc)

**Comment by molinla** [comment]
vscode works fine on this project, because its biome extension will use itself `biome.exe` not in this project installed `@biomejs/biome`


Here is vscode biome extension file path:

![image](https://github.com/user-attachments/assets/d1897184-06ba-4916-9f6f-3bb7f4dfc832)

**Comment by denbezrukov** [comment]
@molinla 
Could you try please to use the night build?
https://plugins.jetbrains.com/plugin/22761-biome/versions/nightly

**Comment by molinla** [comment]
pretty sad, after installed nightly plugin, its not working in auto config mode it will still remove all my current file codes. and manuly set `biome.exe` now not working too.
preview:

https://github.com/user-attachments/assets/5c40b2e6-5a36-4c9b-b428-e0c5bc6f3b13

here is my monorepo project struct:

root
├── README.md
├── apps
│   ├── crdt.app
│   ├── oss-service.app
│   ├── platform.app
│   ├── uidl-crdt.app
│   ├── user-login.app
│   └── workspace.app
├── biome.jsonc
├── node_modules
│   ├── @antfu
│   ├── @biomejs
│   ├── @eslint
│   ├── @eslint-community
│   ├── @eslint-react
│   ├── @stylistic
│   ├── @types
│   ├── @typescript-eslint
│   ├── @vitest
│   ├── eslint
│   ├── eslint-config-prettier
│   ├── eslint-plugin-prettier
│   ├── eslint-scope
│   ├── eslint-visitor-keys
│   ├── husky
│   ├── prettier
│   └── prettier-linter-helpers
├── package.json
├── packages
│   ├── base
│   ├── core
│   ├── platform
│   └── standard
├── pnpm-lock.yaml
├── pnpm-workspace.yaml
└── tools
    ├── figma-import
    └── lint

**Comment by denbezrukov** [comment]
@molinla
Could you create and share a minimal reproducible repo example? 
It'll be very useful🙏

**Comment by molinla** [comment]
> @molinla Could you create and share a minimal reproducible repo example? It'll be very useful🙏

here is minimal reproducible repo example: https://github.com/molinla/biome-format-testing

thansk for you help

**Comment by denbezrukov** [comment]
@molinla 
Thank you!

I'll try to investigate further. I need to find a computer with Windows. Do I understand correctly that it always occurs, regardless of what you do? I mean, even if you only turn on formatting or only fixing, it still happens?

**Comment by molinla** [comment]
yes, for now when my computer is windows, it always remove all my file codes when i turnning auto detect biome config, and turn on auto fix

**Comment by denbezrukov** [comment]
Could you try the last nightly build? 
https://plugins.jetbrains.com/plugin/22761-biome/versions/nightly

**Comment by denbezrukov** [comment]
If 1.6.0 doesn't fix the issue feel free to reopen.

**Comment by molinla** [comment]
> If 1.6.0 doesn't fix the issue feel free to reopen.

it's works well on my windows! :)
